postgres/datadog_checks/postgres/postgres.py

```diff
@@ -365,37 +378,47 @@ def _is_9_4_or_above(self, key, db):
     def _is_10_or_above(self, key, db):
         return self._is_above(key, db, [10, 0, 0])
 
-    def _get_instance_metrics(self, key, db, database_size_metrics, with_default):
-        """Use either COMMON_METRICS or COMMON_METRICS + NEWER_92_METRICS
-        depending on the postgres version.
-        Uses a dictionnary to save the result for each instance
+    def _get_instance_metrics(self, key, db, database_size_metrics, collect_default_db):
+        """
+        Add NEWER_92_METRICS to the default set of COMMON_METRICS when server
+        version is 9.2 or later.
+
+        Store the list of metrics in the check instance to avoid rebuilding it at
+        every collection cycle.
+
+        In case we have multiple instances pointing to the same postgres server
+        monitoring different databases, we want to collect server metrics
+        only once. See https://github.com/DataDog/dd-agent/issues/1211
         """
-        # Extended 9.2+ metrics if needed
         metrics = self.instance_metrics.get(key)
 
         if metrics is None:
-            # Hack to make sure that if we have multiple instances that connect to
-            # the same host, port, we don't collect metrics twice
-            # as it will result in https://github.com/DataDog/dd-agent/issues/1211
-            sub_key = key[:2]
-            if sub_key in self.db_instance_metrics:
-                self.instance_metrics[key] = None
```

Comment:
I believe this was actually a bug, there's no point in explicitly set this value to `None`. My guess is OP wanted to skip this `if` block altogether and early-return at line 399 but to do so the value should be assigned to an empty list, not `None`.